File: src/main/java/com/xelita/commons/Identifiable.java
package com.xelita.commons;

/**
 * Define an identifiable object.
 *
 * @author xelita
 */
public interface Identifiable {

    /**
     * Return the object identifier.
     *
     * @return String
     */
    String getId();

    /**
     * Return the type of the object.
     *
     * @return String
     */
    String getType();
}

File: src/main/kotlin/com/xelita/commons/Identifiable.kt
package com.xelita.commons;

import com.fasterxml.jackson.annotation.JsonIgnore

/**
 * Define an identifiable object.
 *
 * @author xelita
 */
interface Identifiable {

    /**
     * Return the object identifier.
     *
     * @return String
     */
    @JsonIgnore
    fun getId(): String;

    /**
     * Return the type of the object.
     *
     * @return String
     */
    @JsonIgnore
    fun getType(): String;
}
